### ISSUE TITLE
Is resampling needed when using EnCodec?

### ISSUE THREAD

**Issue by m-pana** [lucidrains/audiolm-pytorch#218]
Hi,
Thanks for this amazing repository.

I'm starting to play around with the models and I would like to use EnCodec to save me the trouble of training SoundStream (obviously).
I was wondering, how is the sample rate mismatch between EnCodec and HuBERT handled? I see from [fairseq](https://github.com/facebookresearch/fairseq/tree/main/examples/hubert) that HuBERT base was trained on LibriSpeech, which is 16k.

Given that, do I need to use 24k audio or 16k audio? Does the code have some automatic internal resampling that takes care of the mismatch between the two models? I've looked for it, but couldn't find it.

Thanks a lot!

**Comment by lucidrains** [comment]
@m-pana oh hey Michele! thanks and no problem

you caught me at the right time, as I am about to return to some audio work

do you want to see if [this commit](https://github.com/lucidrains/audiolm-pytorch/commit/053bfe8ceee262868d20f36d32e7c2cca20b940e) solves your issue?

**Comment by lucidrains** [comment]
```python
import torch
from audiolm_pytorch import EncodecWrapper

encodec = EncodecWrapper()

x = torch.randn(1, 48000)
out = encodec(x, input_sample_hz = 16000)
```

**Comment by lucidrains** [comment]
@m-pana ohh, actually, if you are training, the dataset should automatically figure out what the sampling frequencies [it needs](https://github.com/lucidrains/audiolm-pytorch/blob/main/audiolm_pytorch/trainer.py#L895), and internally the proper sampled audio is forwarded to the correct module

**Comment by lucidrains** [comment]
@m-pana i think the scenario i have not covered yet is the prompt waveforms on inference - the input sample hz should be specified and it should auto convert for the semantic and acoustic models

maybe we should leave this issue so i resolve it this week

**Comment by m-pana** [comment]
> @m-pana ohh, actually, if you are training, the dataset should automatically figure out what the sampling frequencies [it needs](https://github.com/lucidrains/audiolm-pytorch/blob/main/audiolm_pytorch/trainer.py#L895), and internally the proper sampled audio is forwarded to the correct module

Oh, I had completely missed that!

> @m-pana i think the scenario i have not covered yet is the prompt waveforms on inference - the input sample hz should be specified and it should auto convert for the semantic and acoustic models
> 
> maybe we should leave this issue open so i resolve it this week

Alright, thanks!

**Comment by p0p4k** [comment]
Regarding encodec, can it process batched input of wav files? Where do we mask the wav files?
